contracts/base/LockAndMsgSender.sol

```diff
@@ -1,35 +1,27 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity ^0.8.24;
 
-import {Constants} from '../libraries/Constants.sol';
+import {Locker} from '../libraries/Locker.sol';
 
 contract LockAndMsgSender {
     error ContractLocked();
 
-    address internal constant NOT_LOCKED_FLAG = address(1);
-    address internal lockedBy = NOT_LOCKED_FLAG;
-
+    /// @notice Modifier enforcing a reentrancy lock that allows self-reentrancy
```

Comment:
Can an external party that is not msg.sender trigger self reentrancy?

Reply:
no theres only 1 entry point to the contract so once its locked only the locker has control